src/memory/transfer_context.cpp

```diff
@@ -43,7 +43,7 @@ SharedBuffer* TransferContext::get_buffer(jni::Env& p_env) {
         shared_buffer = SharedBuffer {address, 0};
 #endif
     }
-
+    shared_buffer.rewind();
```

Comment:
This part i find a bit dangerous. If one calls `get` it has the side effect of rewinding the buffer. I do agree that this is the correct way of doing it, but i would rename the method from `get` to `get_and_rewind` or `use` ore something like that.

Reply:
I can do that yes, not a big deal.
I just don't think it dangerous to use as it is, `get_buffer` is a private method so we are not exposing a dangerous method anyway.

Reply:
True